3 - FightClub/Code/FightClub.cs

```diff
@@ -7,17 +8,88 @@ public static class FightClubGame
 	{
 		public static IPowerUp PurchasePowerUp()
 		{
-			throw new NotImplementedException();
-		}
+            var powerUp = new PowerUp
+            {
+                DamageBuff = 1
+            };
+            return powerUp;
+        }
+
+
 
-		public static IPotion PurchasePotion()
+
+        public static IPotion PurchasePotion()
 		{
-			throw new NotImplementedException();
-		}
+            var potion = new Potion
+            {
+                Healing = 2,
+                Type = PotionType.Basic
+            };
+            return potion;
+        }
 
 		public static IFighter HireFighter()
 		{
-			throw new NotImplementedException();
+            var fighter = new Fighter
+            {
+                Damage = 1,
+                Health = 10
+            };
+            return fighter;
 		}
-	}
+
+        class PowerUp : IPowerUp
+        {
+            public int DamageBuff { get; set; }
+        }
+
+        class Potion : IPotion
+        {
+            public int Healing { get; set; }
+            public PotionType Type { get; set; }
+
+            public void SetType(PotionType potionType) 
+            {
+                if (Enum.Equals(potionType, PotionType.Greater))
+                { 
+                    Healing = 5; 
+                    Type = PotionType.Greater; 
+                }
+            }
```

Comment:
You do not need the Enum.Equals here, that is to much overkill for this. You can actually just do `PotionType.Greater == potionType`